### PR TITLE
XTokens refactor

### PR DESCRIPTION
**Pull request by xlc**
Unified events so there is only one to deal with
Not allow zero amount because apparently it is not allowed https://github.com/paritytech/polkadot/issues/4996

### CODE REVIEW

File: xtokens/src/tests.rs [hunk 1]
```diff
@@ -1048,3 +1048,54 @@ fn specifying_a_non_existent_asset_index_should_fail() {
...
+					)
+						.into()
+				),
+				40,
+			),
+			Error::<para::Runtime>::ZeroAmount
+		);
+	});
+
+	// TODO: should have more tests after https://github.com/paritytech/polkadot/issues/4996
```
Comment (xlc) [inline xtokens/src/tests.rs:1100]: should have more test to ensure MultiAsset is valid (fungible & non zero amount) after https://github.com/paritytech/polkadot/issues/4996

**Review by shaunxw — APPROVED** [review]
LGTM
